File: web/models/ai.py
from pydantic import BaseModel
from typing import List, Dict, Any, Optional
from datetime import datetime

class AnalysisRequest(BaseModel):
    batch_id: str
    analysis_types: List[str]

class AnalysisResponse(BaseModel):
    analysis_id: str
    batch_id: str
    analysis_types: List[str]
    results: Dict[str, Any]
    confidence_score: float
    started_at: datetime
    completed_at: datetime
    status: str
    error_message: Optional[str] = None

class PatternAnalysisResponse(BaseModel):
    pattern_id: str
    pattern_type: str
    pattern: str
    frequency: int
    confidence: float
    examples: List[str]
    suggested_rule: Optional[Dict[str, str]] = None
    impact_assessment: str
    discovered_at: str

class FeedbackRequest(BaseModel):
    product_id: str
    original_description: str
    generated_description: str
    feedback_text: str
    rating: int
    feedback_type: str
    suggestions: Optional[str] = None

class FeedbackResponse(BaseModel):
    feedback_id: str
    status: str
    processing_result: Any
    message: str

class ConfidenceAnalysisResponse(BaseModel):
    analysis_id: str
    batch_id: Optional[str]
    time_period_days: int
    overall_confidence: float
    confidence_distribution: Dict[str, float]
    confidence_trends: List[Any]
    low_confidence_patterns: List[Any]
    improvement_suggestions: List[str]
    generated_at: datetime

File: frontend/src/types/ai.ts
export interface AnalysisRequest {
    batch_id: string;
    analysis_types: string[];
  }
  
  export interface AnalysisResponse {
    analysis_id: string;
    batch_id: string;
    analysis_types: string[];
    results: Record<string, any>;
    confidence_score: number;
    started_at: string;
    completed_at: string;
    status: string;
    error_message?: string;
  }
  
  export interface PatternAnalysisResponse {
    pattern_id: string;
    pattern_type: string;
    pattern: string;
    frequency: number;
    confidence: number;
    examples: string[];
    suggested_rule?: {
      rule_type: string;
      pattern: string;
      replacement: string;
    };
    impact_assessment: string;
    discovered_at: string;
  }
  
  export interface FeedbackRequest {
    product_id: string;
    original_description: string;
    generated_description: string;
    feedback_text: string;
    rating: number;
    feedback_type: string;
    suggestions?: string;
  }
  
  export interface FeedbackResponse {
    feedback_id: string;
    status: string;
    processing_result: any;
    message: string;
  }
  
  export interface ConfidenceAnalysisResponse {
    analysis_id: string;
    batch_id?: string;
    time_period_days: number;
    overall_confidence: number;
    confidence_distribution: Record<string, number>;
    confidence_trends: any[];
    low_confidence_patterns: any[];
    improvement_suggestions: string[];
    generated_at: string;
  }